### PR TITLE
#261 Make routes static

### PR DESCRIPTION
**Pull request by anilgursel**

### CODE REVIEW

File: squbs-unicomplex/src/main/scala/org/squbs/unicomplex/streaming/ServiceRegistry.scala [hunk 1]
```diff
@@ -69,6 +71,8 @@ class ServiceRegistry(val log: LoggingAdapter) extends ServiceRegistryBase[Path]
     implicit val am = ActorMaterializer()
     import context.system
 
+    val agent = listenerRoutes(name)
+    Await.ready(agent.future(), UnicomplexBoot.defaultStartupTimeout.duration)
```
Comment (akara) [inline squbs-unicomplex/src/main/scala/org/squbs/unicomplex/streaming/ServiceRegistry.scala]: I'd not put blocking code here. If you really wanted to, you can pass agent.future back and compose the future to return the ready signal from `Unicomplex` to `CubeSupervisor` instead.

Comment (anilgursel) [inline squbs-unicomplex/src/main/scala/org/squbs/unicomplex/streaming/ServiceRegistry.scala]: You mean one of the following?

- https://github.com/paypal/squbs/blob/master/squbs-unicomplex/src/main/scala/org/squbs/unicomplex/UnicomplexBoot.scala#L409

- https://github.com/paypal/squbs/blob/master/squbs-unicomplex/src/main/scala/org/squbs/unicomplex/UnicomplexBoot.scala#L596

If so, then I won't have access to agent in any of those places (unless I pass that around, which might make it more complex).

How about changing the logic here: https://github.com/paypal/squbs/blob/master/squbs-unicomplex/src/main/scala/org/squbs/unicomplex/Unicomplex.scala#L397 from 

```scala
case StartListener(name, conf) => // Sent from Bootstrap to start the web service infrastructure.

      Try { serviceRegistry.startListener(name, conf, notifySender = sender()) } match {
        case Success(startupBehavior) =>
          context.become(startupBehavior orElse {
            case HttpBindSuccess =>
              if (serviceRegistry.isListenersBound) updateSystemState(checkInitState())
              context.unbecome()
              unstashAll()
            case HttpBindFailed =>
              updateSystemState(checkInitState())
              context.unbecome()
              unstashAll()

            case _ => stash()
          },
            discardOld = false)

        case Failure(t) => updateSystemState(checkInitState())
      }
```

to (please note, this is just pseudo code, haven't covered all cases and not tested and even compiled):

```scala
case StartListener(name, conf) => // Sent from Bootstrap to start the web service infrastructure.
      serviceRegistry.listenerRoutes(name) map { case Success(_) => ReadyToStartListener(name, conf) } pipeTo self

case ReadyToStartListener(name, conf) => 
      Try { serviceRegistry.startListener(name, conf, notifySender = sender()) } match {
        case Success(startupBehavior) =>
          context.become(startupBehavior orElse {
            case HttpBindSuccess =>
              if (serviceRegistry.isListenersBound) updateSystemState(checkInitState())
              context.unbecome()
              unstashAll()
            case HttpBindFailed =>
              updateSystemState(checkInitState())
              context.unbecome()
              unstashAll()

            case _ => stash()
          },
            discardOld = false)

        case Failure(t) => updateSystemState(checkInitState())
      }
```

File: squbs-unicomplex/src/main/scala/org/squbs/unicomplex/ServiceRegistryBase.scala [hunk 2]
```diff
@@ -19,31 +19,29 @@ package org.squbs.unicomplex
...
   private[unicomplex] def prepListeners(listenerNames: Iterable[String])(implicit context: ActorContext) {
-    import context.dispatcher
     listenerRoutes = listenerNames.map { listener =>
-      listener -> Agent[Seq[(A, ActorWrapper, PipelineSetting)]](Seq.empty)
+      listener -> Seq.empty[(A, ActorWrapper, PipelineSetting)]
     }.toMap
 
     import org.squbs.unicomplex.JMX._
-    register(new ListenerBean(listenerRoutes), prefix + listenersName)
+    register(new ListenerBean(() => listenerRoutes), prefix + listenersName)
```
Comment (akara) [inline squbs-unicomplex/src/main/scala/org/squbs/unicomplex/ServiceRegistryBase.scala]: Only one question. Why do we decide to use the function signature here? Are we expecting any lazy behavior here? And if so, would a by-name parameter not be more natural here?

Comment (anilgursel) [inline squbs-unicomplex/src/main/scala/org/squbs/unicomplex/ServiceRegistryBase.scala]: At the time we register `ListenerBean` in this line, `listenerRoutes` is not ready yet.  That's the reason we have the function signature.

**Pull request merged**